components/automate-ui/src/app/entities/environments/environment.model.ts

```diff
@@ -1,3 +1,38 @@
+import { merge } from 'lodash';
+
 export interface Environment {
   name: string;
+  chef_type: string;
+  description: string;
+  json_class: string;
+  default_attributes: string;
+  override_attributes: string;
+  run_list: string[];
+  cookbook_versions: CookbookVersion[];
+}
+
+export interface CookbookVersion {
+  name: string;
+  version: string;
+}
+
+export class EnvironmentAttributes {
+  default_attributes: string;
+  override_attributes: string;
+  all: string;
+  constructor(resp: RespEnvironmentAttributes) {
+    this.default_attributes = resp.default_attributes &&
+      JSON.parse(resp.default_attributes) || {};
+    this.override_attributes = resp.override_attributes &&
+      JSON.parse(resp.override_attributes) || {};
```

Comment:
Your line breaks actually imply different precedence than actual precedence; consider adding parentheses and adjusting as shown to disambiguate to the reader.
```suggestion
    this.default_attributes =
      (resp.default_attributes && JSON.parse(resp.default_attributes)) || {};
    this.override_attributes =
      (resp.override_attributes && JSON.parse(resp.override_attributes)) || {};
```